lib/vagrant-libvirt/config.rb

```diff
@@ -949,6 +957,23 @@ def validate(machine)
           end
         end
 
+
+        if @qemu_use_agent == true
+          # if qemu agent is used to optain domain ip configuration, at least
+          # one qemu channel has to be configured. As there are various options,
+          # error out and leave configuration to the user
+          err = true
+          machine.provider_config.channels.each { |channel|
+            if channel[:target_name].start_with?("org.qemu.guest_agent")
+              err = nil
+            end
+          }
+          if err == true
+            errors << "qemu agent option enabled, but no qemu agent channel configured: please add at least one qemu agent channel to vagrant config"
+          end
```

Comment:
I think I'd be inclined to use the `.any?` syntax here rather than `.each`
```suggestion
          unless machine.provider_config.channels.any? { |channel| channel[:target_name].start_with?("org.qemu.guest_agent") }
            errors << "qemu agent option enabled, but no qemu agent channel configured: please add at least one qemu agent channel to vagrant config"
          end
```